src/goto-instrument/nondet_static.cpp

```diff
@@ -136,3 +140,64 @@ void nondet_static(goto_modelt &goto_model)
   const namespacet ns(goto_model.symbol_table);
   nondet_static(ns, goto_model.goto_functions);
 }
+
+/// Second main entry point of the module. Nondeterministically initializes
+/// global scope variables, except for constants (such as string literals, final
+/// fields), internal variables (such as CPROVER and symex variables,
+/// language specific internal variables) and variables passed to except_value.
+/// \param [out] goto_model: Existing goto-model to be updated.
+/// \param except_values: list of symbol names that should not be updated.
+void nondet_static(
+  goto_modelt &goto_model,
+  const optionst::value_listt &except_values)
+{
+  const namespacet ns(goto_model.symbol_table);
+  std::set<std::string> to_exclude;
+
+  for(auto const &except : except_values)
+  {
+    bool prefix_found = false;
+    auto it = except.find(":");
+
+    if(it != std::string::npos)
+    {
+      prefix_found = true;
+      to_exclude.insert(except);
+      if(except.length() > 2 && except[0] == '.' && except[1] == '/')
+      {
+        to_exclude.insert(except.substr(2, except.length() - 2));
+      }
+      else
+      {
+        to_exclude.insert("./" + except);
```

Comment:
I don't really understand this piece of code, if `except` is "./path" we insert "path" and if it is "path" we insert "./path"?

Reply:
Yes. The location path returned by the namespace symbol after lookup sometimes starts with `./` and sometimes not. To make the lookup more deterministic, I add the prefix eventually to the passed parameter or remove it. So the exclude list always contains `path` and `./path` as an excluded location.

Reply:
Ok I see. Maybe add a comment about that. Ideally I would expect the path to be stored in some canonical form (but I guess it's alright for now)

Reply:
Ok I see. Maybe add a comment about that. Ideally I would expect the path to be stored in some canonical form (but I guess it's alright for now)